src/parsers/json/Parser.cpp

```diff
@@ -933,6 +933,23 @@ Grid3 Parser::readGrids(const json& j) const {
         std::string folder = this->filename.getFolder();
         std::string projectName = this->filename.getProjectName();
         return buildGridFromFile(folder + projectName + ".grid.json");
+    } else if (gridType.compare("rectilinear") == 0) {
+        std::map<std::string, std::vector<double>> planes;
+        for (const auto& label: { "xs", "ys", "zs"}) {
+            auto dir{ g.find(label) };
+            if (dir != g.end()) {
+                auto plane{ dir->get<std::vector<double>>() };
+                if (plane.empty()) {
+                    throw std::logic_error("Grid file had empty positions in at least one direction");
+                }
+                planes.emplace(label, plane);
+            }
+            else {
```

Comment:
I think it would be better to check that dir == g.end() and throw. The code inside the if (dir != g.end()) can be out of the if and lowers complexity.

Reply:
Indeed. I find the code perfectly fine. 

My comments would go in the same direction. Since we are here, we can, just for this `readGrids` method:
- Remove unnecessary indent of `if` - `else` that are mutually exclusive. As a general rule, if you have a `if` and `else` ending both in a `return` statement, you can just check one condition and place the `return` at the end. 
- If a variable is declared and passed through a method directly, as lines 951 and  952, I tend to have it directly in one place. I don't know if that could arise a issue in C++'s land regarding of destruction when exiting from the method itself...
- Try to use possitive checking. They are easier to read (Line 940 or inside `nativeGiD`)

Besides that, I would go a bit deeper in the test and check the exact positions of the 2, 3, and 4 values for x, y and z.

I would provide examples:

![imagen](https://user-images.githubusercontent.com/11597234/198572384-cfad4f91-e497-4b2e-95f0-d3ed4ba4bfc6.png)

![imagen](https://user-images.githubusercontent.com/11597234/198572441-3fbcbe91-f2ba-4175-b0ec-f252026000cd.png)

![imagen](https://user-images.githubusercontent.com/11597234/198572504-f224d53e-935f-479a-9de6-41a9d9234840.png)